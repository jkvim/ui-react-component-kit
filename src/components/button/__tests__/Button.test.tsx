import * as React from 'react';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { Button } from '../';

describe('Button', () => {

  describe('props', () => {

    const mockCallback = jest.fn();

    const wrapper = mount(
      <Button type="primary" disabled={true} onClick={mockCallback}>Hello</Button>);

    it('should match snapshot', () => {
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it('should pass `type` in props', () => {
      expect(wrapper.props().type).toBe('primary');
    });

    it('should pass `disabled` in props', () => {
      expect(wrapper.props().disabled).toBe(true);
    });

    it('should pass `onClick` in props', () => {
      expect(wrapper.props().onClick).toBe(mockCallback);
    });

    it('should pass `children` in props', () => {
      expect(wrapper.props().children).toBe('Hello');
    });
  });

  it('snapshot basic', () => {
    const mockCallback = jest.fn();
    const wrapper = mount(<Button type="primary" onClick={mockCallback}>Hello</Button>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('snapshot disabled', () => {
    const mockCallback = jest.fn();
    const wrapper = mount(<Button type="primary" disabled={true} onClick={mockCallback}>Hello</Button>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('should call callback on click', () => {
    const mockCallback = jest.fn();
    const wrapper = mount(<Button type="primary" onClick={mockCallback}>Hello</Button>);
    wrapper.simulate('click');
    expect(mockCallback).toBeCalled();
  });
});