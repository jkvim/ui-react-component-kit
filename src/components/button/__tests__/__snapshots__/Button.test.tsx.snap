// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button props should match snapshot 1`] = `
<Button
  disabled={true}
  onClick={[MockFunction]}
  type="primary"
>
  <styled.button
    disabled={true}
    onClick={[Function]}
    theme={
      Object {
        "border": Object {
          "radius": Object {
            "edge": "2px",
            "round": "100px",
          },
          "size": "2px",
        },
        "breakpoint": Object {
          "lg": "992px",
          "md": "768px",
          "sm": "576px",
          "xl": "1200px",
          "xs": "0px",
        },
        "color": Object {
          "background": "#ffffff",
          "border": "#2fa4e7",
          "checked": "#2fa4e7",
          "danger": "#c71c22",
          "default": "#2fa4e7",
          "disabled": "#acacac",
          "error": "#c71c22",
          "focus": "#54b4eb",
          "font": "#333333",
          "hover": "#178acc",
          "info": "#04519b",
          "primary": "#2fa4e7",
          "secondary": "#e9ecef",
          "success": "#73a839",
          "tertiary": "#343a40",
          "warning": "#dd5600",
        },
        "font": Object {
          "size": Object {
            "default": "1.8rem",
          },
          "weight": Object {
            "default": 100,
          },
        },
      }
    }
    type="primary"
  >
    <button
      className="sc-bdVaJa hAcUsb"
      disabled={true}
      onClick={[Function]}
      type="primary"
    >
      Hello
    </button>
  </styled.button>
</Button>
`;

exports[`Button snapshot disabled 1`] = `
<Button
  disabled={true}
  onClick={[MockFunction]}
  type="primary"
>
  <styled.button
    disabled={true}
    onClick={[Function]}
    theme={
      Object {
        "border": Object {
          "radius": Object {
            "edge": "2px",
            "round": "100px",
          },
          "size": "2px",
        },
        "breakpoint": Object {
          "lg": "992px",
          "md": "768px",
          "sm": "576px",
          "xl": "1200px",
          "xs": "0px",
        },
        "color": Object {
          "background": "#ffffff",
          "border": "#2fa4e7",
          "checked": "#2fa4e7",
          "danger": "#c71c22",
          "default": "#2fa4e7",
          "disabled": "#acacac",
          "error": "#c71c22",
          "focus": "#54b4eb",
          "font": "#333333",
          "hover": "#178acc",
          "info": "#04519b",
          "primary": "#2fa4e7",
          "secondary": "#e9ecef",
          "success": "#73a839",
          "tertiary": "#343a40",
          "warning": "#dd5600",
        },
        "font": Object {
          "size": Object {
            "default": "1.8rem",
          },
          "weight": Object {
            "default": 100,
          },
        },
      }
    }
    type="primary"
  >
    <button
      className="sc-bdVaJa hAcUsb"
      disabled={true}
      onClick={[Function]}
      type="primary"
    >
      Hello
    </button>
  </styled.button>
</Button>
`;

exports[`Button snapshot primary 1`] = `
<Button
  onClick={[MockFunction]}
  type="primary"
>
  <styled.button
    onClick={[Function]}
    theme={
      Object {
        "border": Object {
          "radius": Object {
            "edge": "2px",
            "round": "100px",
          },
          "size": "2px",
        },
        "breakpoint": Object {
          "lg": "992px",
          "md": "768px",
          "sm": "576px",
          "xl": "1200px",
          "xs": "0px",
        },
        "color": Object {
          "background": "#ffffff",
          "border": "#2fa4e7",
          "checked": "#2fa4e7",
          "danger": "#c71c22",
          "default": "#2fa4e7",
          "disabled": "#acacac",
          "error": "#c71c22",
          "focus": "#54b4eb",
          "font": "#333333",
          "hover": "#178acc",
          "info": "#04519b",
          "primary": "#2fa4e7",
          "secondary": "#e9ecef",
          "success": "#73a839",
          "tertiary": "#343a40",
          "warning": "#dd5600",
        },
        "font": Object {
          "size": Object {
            "default": "1.8rem",
          },
          "weight": Object {
            "default": 100,
          },
        },
      }
    }
    type="primary"
  >
    <button
      className="sc-bdVaJa hAcUsb"
      onClick={[Function]}
      type="primary"
    >
      Hello
    </button>
  </styled.button>
</Button>
`;

exports[`Button snapshot secondary 1`] = `
<Button
  onClick={[MockFunction]}
  type="secondary"
>
  <styled.button
    onClick={[Function]}
    theme={
      Object {
        "border": Object {
          "radius": Object {
            "edge": "2px",
            "round": "100px",
          },
          "size": "2px",
        },
        "breakpoint": Object {
          "lg": "992px",
          "md": "768px",
          "sm": "576px",
          "xl": "1200px",
          "xs": "0px",
        },
        "color": Object {
          "background": "#ffffff",
          "border": "#2fa4e7",
          "checked": "#2fa4e7",
          "danger": "#c71c22",
          "default": "#2fa4e7",
          "disabled": "#acacac",
          "error": "#c71c22",
          "focus": "#54b4eb",
          "font": "#333333",
          "hover": "#178acc",
          "info": "#04519b",
          "primary": "#2fa4e7",
          "secondary": "#e9ecef",
          "success": "#73a839",
          "tertiary": "#343a40",
          "warning": "#dd5600",
        },
        "font": Object {
          "size": Object {
            "default": "1.8rem",
          },
          "weight": Object {
            "default": 100,
          },
        },
      }
    }
    type="secondary"
  >
    <button
      className="sc-bdVaJa hFxAHV"
      onClick={[Function]}
      type="secondary"
    >
      Hello
    </button>
  </styled.button>
</Button>
`;

exports[`Button snapshot tertiary 1`] = `
<Button
  onClick={[MockFunction]}
  type="tertiary"
>
  <styled.button
    onClick={[Function]}
    theme={
      Object {
        "border": Object {
          "radius": Object {
            "edge": "2px",
            "round": "100px",
          },
          "size": "2px",
        },
        "breakpoint": Object {
          "lg": "992px",
          "md": "768px",
          "sm": "576px",
          "xl": "1200px",
          "xs": "0px",
        },
        "color": Object {
          "background": "#ffffff",
          "border": "#2fa4e7",
          "checked": "#2fa4e7",
          "danger": "#c71c22",
          "default": "#2fa4e7",
          "disabled": "#acacac",
          "error": "#c71c22",
          "focus": "#54b4eb",
          "font": "#333333",
          "hover": "#178acc",
          "info": "#04519b",
          "primary": "#2fa4e7",
          "secondary": "#e9ecef",
          "success": "#73a839",
          "tertiary": "#343a40",
          "warning": "#dd5600",
        },
        "font": Object {
          "size": Object {
            "default": "1.8rem",
          },
          "weight": Object {
            "default": 100,
          },
        },
      }
    }
    type="tertiary"
  >
    <button
      className="sc-bdVaJa cfUbqE"
      onClick={[Function]}
      type="tertiary"
    >
      Hello
    </button>
  </styled.button>
</Button>
`;
